src/main/java/com/redislabs/redisgraph/graph_entities/Point.java

```diff
@@ -0,0 +1,62 @@
+package com.redislabs.redisgraph.graph_entities;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class represents a (geographical) point in the graph.
+ */
+public final class Point {
+
+    private static final double EPSILON = 1e-5;
+
+    private final double latitude;
+    private final double longitude;
+
+    /**
+     * @param latitude
+     * @param longitude 
+     */
+    public Point(double latitude, double longitude) {
+        this.latitude = latitude;
+        this.longitude = longitude;
+    }
+
+    /**
+     * @param values {@code [latitude, longitude]}
+     */
+    public Point(List<Double> values) {
+        if (values == null || values.size() != 2) {
+            throw new IllegalArgumentException("Point requires two doubles.");
+        }
+        this.latitude = values.get(0);
+        this.longitude = values.get(1);
+    }
+
+    public double getLatitude() {
+        return latitude;
+    }
+
+    public double getLongitude() {
+        return longitude;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) return true;
+        if (!(other instanceof Point)) return false;
+        Point o = (Point) other;
+        return Math.abs(latitude - o.latitude) < EPSILON &&
+                Math.abs(longitude - o.longitude) < EPSILON;
```

Comment:
I think we should find the mapping format (datum) resolution and infer `EPSILON` from it, otherwise, we can get wrong results here

Reply:
@DvirDukhan I'd prefer to avoid that. The difference is already very low and datum resolution would just bring more precision issue. More importantly, we are not calculating the distance here. We are just trying to check if the latitudes and longitudes (both plural) can be considered same.